httpx/dispatch/http11.py

```diff
@@ -166,6 +166,9 @@ def __init__(
                     )
                 except OSError:  # pragma: nocover
                     data = b""
+                if self.h11_state.their_state == h11.SEND_RESPONSE and data == b"":
+                    # the server closed a keep-alive connection
+                    raise ConnectionClosedByRemote
```

Comment:
We can just use `NotConnected` here. It's there to handle the exact same case, it's just that we might *either* see it on sending data, or on receiving data, and we were only dealing with one of those two cases prior to this PR.

Reply:
We'll need to think about the naming. The exception needs to be the subset of connection closes that occur *only* at the start of reading/writing a response.

There's also something worth digging into here, which isn't clear to me. Why isn't the case being triggered by `_send_request` in the first place before we get here? We're writing to the network, and calling drain, so I'd expect to see the exception already.